clients/app/src/views/screens/ProfileList/ProfileList.tsx

```diff
@@ -4,35 +4,67 @@ import I18n from 'i18n/i18n';
 import Container from 'views/components/base/Container';
 import { ProfileListTypeEnum } from './';
 import { IProfileListProps } from './types';
+import buildEventMembersStore from 'stores/eventMembers';
+import buildEventCatchesStore from 'stores/eventCatches';
+import buildRoomCatchesStore from 'stores/roomCatches';
+import buildMatchCatchesStore from 'stores/matchCatches';
 
 const ProfileListContainer = React.lazy(
   () => import('./containers/ProfileListContainer')
 );
 
+const defaultLimitation = 10;
+
 const ProfileList: React.FC<IProfileListProps> = ({ route }) => {
-  const { id, type } = route.params;
+  const { id, type, limit = defaultLimitation } = route.params;
 
   const [store, title] = React.useMemo(() => {
     switch (type) {
       case ProfileListTypeEnum.fanClubMembers:
         return [buildFanClubMembersStore(), I18n.t('fanClubs-fans')];
       case ProfileListTypeEnum.eventMembers:
-        return [buildFanClubMembersStore(), I18n.t('common-going')];
-      case ProfileListTypeEnum.eventCatch:
-        return [buildFanClubMembersStore(), I18n.t('common-catches')];
+        return [buildEventMembersStore(), I18n.t('common-going')];
+      case ProfileListTypeEnum.eventCatches:
+        return [buildEventCatchesStore(), I18n.t('catches-event')];
+      case ProfileListTypeEnum.roomCatches:
+        return [buildRoomCatchesStore(), I18n.t('catches-room')];
+      case ProfileListTypeEnum.matchCatches:
+        return [buildMatchCatchesStore(), I18n.t('catches-match')];
     }
   }, [type]);
 
   const { list: storeList } = store.useSelector('list');
 
   React.useEffect(() => {
-    // storeList.getAll({ fanClubId: id });
+    switch (type) {
+      case ProfileListTypeEnum.fanClubMembers:
+        storeList.updateFilter({ fanClubId: id, limit: limit });
```

Comment:
can't we move filter name to upper React.useMemo, and replace this switch with one row?

Reply:
no, we can't move React.useEffect logic into React.useMemo, because inside of useEffect we use store `storeList` which is result of hook `store.useSelector()`, we can't call hook inside of useMemo,

but I change switch logic of useEffect. will push soon

Reply:
I didn't say move React.useEffect, just filter name.